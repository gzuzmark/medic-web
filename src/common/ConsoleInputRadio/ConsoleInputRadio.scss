@import "../../assets/styles/01-settings/settings";

.ConsoleInputRadio {
  cursor: pointer;
  display: flex;
  font-size: 22px;
  margin-bottom: 12px;
  padding-left: 25px;
  position: relative;
  &--input {
    cursor: pointer;
    opacity: 0;
    position: absolute;
    &:checked {
      &~ .ConsoleInputRadio--radio {
        background-color: white;
        &:after {
          background: #4a4a4a;
        }
      }
    }
  }
  &--radio {
    background-color: white;
    border: 1px solid #ccd6dc;
    border-radius: 50%;
    height: 22px;
    left: 0;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    width: 22px;
    &:after {
      background: white;
      border-radius: 50%;
      content: "";
      height: 10px;
      left: 50%;
      position: absolute;
      top: 50%;
      transform: translate(-50%, -50%);
      transition: background ease-in 0.2s;
      width: 10px;
    }
  }
  user-select: none;
}
