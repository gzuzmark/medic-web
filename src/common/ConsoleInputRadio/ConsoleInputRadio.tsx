import * as React from 'react';
import { Text } from '../ConsoleText';
import './ConsoleInputRadio.scss';

interface IPropsConsoleInputRadio {
    title: string;
    name: string;
    checked: boolean;
    value: string;
    onChange: (event?: any) => void
}
const ConsoleInputRadio: React.StatelessComponent<IPropsConsoleInputRadio> = (props) => {
    return (
        <label className="ConsoleInputRadio">
            <Text style={{fontSize: 16}}>{props.title}</Text>
            <input className="ConsoleInputRadio--input"
                   type="radio"
                   name={props.name}
                   value={props.value}
                   checked={props.checked}
                   onChange={props.onChange}/>
            <span className="ConsoleInputRadio--radio">&nbsp;</span>
        </label>
    );
};

export default ConsoleInputRadio;