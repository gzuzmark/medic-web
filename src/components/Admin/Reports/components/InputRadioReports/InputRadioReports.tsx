import * as React from "react";
import {ReportType} from "../../../../../beans/ReportRequest.bean";
import ConsoleInputRadio from "../../../../../common/ConsoleInputRadio/ConsoleInputRadio";
import FormColumn from "../../../../../common/FormRow/components/FormColumn/FormColumn";
import FormRow from "../../../../../common/FormRow/FormRow";
import {IInputRadioReports} from "../../Reports";

interface IPropsInputRadioReports {
    name: string;
    type: ReportType;
    inputs: IInputRadioReports[];
    updateState: (params: object) => void;
}

const InputRadioReports: React.FC<IPropsInputRadioReports> = (props) => {
    const onChange = (event: any) => {
        props.updateState({[props.name]: event.target.value})
    };

    return (
        <FormRow style={{marginTop: 20}} columns={props.inputs.map((element, index) => {
            return (
                <FormColumn key={`InputRadioReports_${index}`}  width={props.inputs.length}>
                    <ConsoleInputRadio
                        title={element.title}
                    attrs={{
                        checked:props.type === element.value,
                        name:props.name,
                        onChange,
                        value:element.value
                    }}/>
                </FormColumn>
            )
        })}/>
    );
};


export default InputRadioReports;
