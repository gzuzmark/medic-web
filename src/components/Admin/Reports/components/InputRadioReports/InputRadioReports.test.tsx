import { mount } from 'enzyme';
import * as React from 'react';
import {REPORT_SESSIONS, REPORT_STUDENTS, ReportType} from "../../../../../beans/ReportRequest.bean";
import ConsoleInputRadio from "../../../../../common/ConsoleInputRadio/ConsoleInputRadio";
import InputRadioReports from "./InputRadioReports";


describe('InputRadioReports Test',() => {
    let props: any;
    let mountedInputRadioReports: any;
    const getComponent = () => {
        if (!mountedInputRadioReports) {
            mountedInputRadioReports = mount(
                <InputRadioReports {...props} />
            );
        }
        return mountedInputRadioReports;
    };

    beforeEach(() => {
        props = {
            inputs: [],
            name: 'demo',
            type: '' as ReportType,
            updateState: undefined
        };
        mountedInputRadioReports = undefined;
    });

    it("render: render InputRadioReports", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });

    it("render: render InputRadioReports with two inputs", () => {
        props = {
            inputs: [{
                title: 'Reporte de sesiones',
                value: REPORT_SESSIONS
            }, {
                title: 'Reporte de alumnos',
                value: REPORT_STUDENTS
            }],
            name: 'demo',
            type: '' as ReportType,
            updateState: undefined
        };
        const component = getComponent();
        const inputs = component.find(ConsoleInputRadio).length;
        expect(inputs).toEqual(2)
    });


    it("events: on change updateState is called", () => {
        const change = jasmine.createSpy('change');
        props = {
            inputs: [{
                title: 'Reporte de sesiones',
                value: REPORT_SESSIONS
            }, {
                title: 'Reporte de alumnos',
                value: REPORT_STUDENTS
            }],
            name: 'demo',
            type: '' as ReportType,
            updateState: change
        };
        const value = {event: {target: {value: REPORT_SESSIONS}}};
        getComponent().find('input').first().simulate('change', value);
        expect(change).toHaveBeenCalled();
    });
});