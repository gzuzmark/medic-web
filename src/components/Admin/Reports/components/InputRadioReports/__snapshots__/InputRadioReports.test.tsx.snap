// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadioReports Test render: render InputRadioReports 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <InputRadioReports
    inputs={Array []}
    name="demo"
    type=""
    updateState={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "inputs": Array [],
      "name": "demo",
      "type": "",
      "updateState": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "columns": Array [],
        "style": Object {
          "marginTop": 20,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="FormRow"
          style="margin-top: 20px;"
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "FormRow",
          "style": Object {
            "marginTop": 20,
          },
        },
        "ref": null,
        "rendered": Array [
          Array [],
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "inputs": Array [],
        "name": "demo",
        "type": "",
        "updateState": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "columns": Array [],
          "style": Object {
            "marginTop": 20,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="FormRow"
            style="margin-top: 20px;"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "FormRow",
            "style": Object {
              "marginTop": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
